wme_junctionangle.user.js

```diff
@@ -502,9 +529,28 @@ function run_ja() {
 					return ja_routing_type.EXIT;
 				}
 			}
+			
+			//wlodek76: FIXING KEEP LEFT/RIGHT regarding to left most segment
+			//WIKI WAZE: When there are more than two segments less than 45.04°, only the left most segment will be KEEP LEFT, all the rest will be KEEP RIGHT
+			
+			//wlodek76: KEEP LEFT/RIGHT overlapping case
+			//WIKI WAZE: If the left most segment is overlapping another segment, it will also be KEEP RIGHT.
+			if (mostleftangle!=null && mostleftangle2!=null) {
+				var overlapped_angle = Math.abs(mostleftangle[0] - mostleftangle2[0]);
+				
+				// If two top most left angles are close < 2 degree they are overlapped.
+				// Method of recognizing overlapped segment by server is unknown for me yet, I took this from WME Validator information about this.
+				// TODO: verify overlapping check on the side of routing server.
+				if (overlapped_angle < 2.0) mostleftangle = null;
+			}
+			
+			if (mostleftangle != null && mostleftangle[1] == s_out_id) {

```

Comment:
Just a thought, is the logic the same for left hand driving countries also? Or would only the rightmost give a "keep right" and all the rest "keep left"